### DISCUSSION TITLE
[ZK Stack] - how to update squencer's code.

### DISCUSSION THREAD

**Discussion by SuccinctPaul** [zkSync-Community-Hub/zksync-developers#619] [ZK Stack]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.75.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

As the zksync-era's updated frequnctly. Once the node is deployed at some version. So wanna know how to upgrade the sequencer to expected one.

### Expected Behavior

Update sequencer code to support more features.

### Current Behavior

....

### Prior Research

- [ ] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
To update the sequencer's code in the zkSync Era, you'll need to follow a structured upgrade process. This process ensures that the updates are secure and do not disrupt the ongoing operations of your node. Here’s a step-by-step guide on how to upgrade the sequencer in the zkSync Era:

### Step-by-Step Guide to Upgrade the Sequencer

1. **Check the Latest Version:**
   - Ensure that you are aware of the latest stable version of the sequencer code. You can check this on the [zkSync Era GitHub repository](https://github.com/matter-labs/zksync-era).

2. **Backup Your Data:**
   - Before proceeding with the upgrade, make sure to back up all relevant data. This includes the current state of the sequencer and any configurations that you might have customized.

3. **Download the Latest Code:**
   - Fetch the latest code from the main branch of the repository. You can do this by cloning the repository or pulling the latest changes if you already have a cloned version.

   ```bash
   git clone https://github.com/matter-labs/zksync-era.git
   # or if you already have the repository
   git pull origin main
   ```

4. **Build the New Version:**
   - Compile the new version of the sequencer. Ensure that your system has the correct Rust version, which in your case is 1.75.0, and that all dependencies are up to date.

   ```bash
   cargo build --release
   ```

5. **Test the New Version:**
   - It is crucial to test the new version in a controlled environment before deploying it to production. This helps in identifying any issues that might not have been caught during development.

6. **Deploy the Upgrade:**
   - Once testing is complete and you are satisfied with the new version, you can deploy it to your production environment. Replace the existing sequencer executable with the new one.

7. **Monitor the Upgrade:**
   - After upgrading, monitor the sequencer closely to ensure that it operates as expected without any issues.

8. **Rollback if Necessary:**
   - If you encounter critical issues that cannot be resolved immediately, be prepared to rollback to the previous version using the backups you created.

### Additional Tips

- **Stay Informed:** Regularly check the repository for updates and announcements related to new features or critical fixes.
- **Community and Support:** If you encounter issues or have questions during the upgrade process, do not hesitate to reach out to the community or seek professional support.

By following these steps, you can successfully upgrade the sequencer in your zkSync Era node to incorporate new features and improvements.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by bxpana** [reply]
@SuccinctPaul were you able to get this figured out? If not let me know and I'll keep trying to get an answer

**Reply by SuccinctPaul** [reply]
Not yes, needs more docs to do it
